### PR TITLE
[DDW-645] Update react-polymorph and implement the search style variables

### PR DESCRIPTION
**Pull request by daniloprates**
This PR updates react-polymorph and implement the search style variables

Some fixes were done in React Polymorph's [#175](https://github.com/input-output-hk/react-polymorph/pull/175)

## Todos

- [x] Update the react-polymorph version
- [x] Define the highlight color to each theme - @alexander-rukin 
- [x] Add the highlight color to the themes
- [x] Fix Sasha's report - @aleksandardjordjeviciohk 
- [x] Update `react-polymorph` once [#170](https://github.com/input-output-hk/react-polymorph/pull/170) is released @aleksandardjordjeviciohk 

## Screenshots

Screen Shot 2021-04-10 at 6 26 07 PM

Screen Shot 2021-04-10 at 6 26 18 PM

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)

---
## Issues summary:

Pending:
- A blank screen and a console error are displayed when I access "Stake pools". Check [video](https://drive.google.com/file/d/1N2Q-BZsGWoeQdT_0jabocEDfyrbS2Sk5/view?usp=sharing).
- This is very minor, but a little inconsistent: when searching for special characters, as "." no highlight is displayed. It does happen with other characters, like "-". Check [video](https://drive.google.com/file/d/1P7ihNXCwSpOgmACglHirvopWyDWdCcQ1/view?usp=sharing).

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by daniloprates** [comment]
@aleksandardjordjeviciohk, here is the @alexander-rukin report:

Wrong size:
image (4)

Border issue:
image (3)

Scroll issue:
image (1)

**Comment by aleksandardjordjeviciohk** [comment]
> @aleksandardjordjeviciohk, here is the @alexander-rukin report:
> 
> Wrong size:
> image (4)
> 
> Border issue:
> image (3)
> 
> Scroll issue:
> image (1)

fixed @alexander-rukin

File: source/renderer/app/components/settings/categories/WalletsSettings.js
Comment (DominikGuzei) [inline source/renderer/app/components/settings/categories/WalletsSettings.js:114]: this shouldn't be necessary as the skins are automatically mapped by react-polymorph (you only need to set it if you want to override with a specific skin)

**Review by DominikGuzei — APPROVED** [review]
code looks great 🎉 just added a small comment

File: source/renderer/app/components/settings/categories/WalletsSettings.js
Comment (aleksandardjordjeviciohk) [reply source/renderer/app/components/settings/categories/WalletsSettings.js:114]: fixed

**Comment by alexander-rukin** [comment]
> > @aleksandardjordjeviciohk, here is the @alexander-rukin report:
> > Wrong size:
> > image (4)
> > Border issue:
> > image (3)
> > Scroll issue:
> > image (1)
> 
> fixed @alexander-rukin

what exactly was fixed? in storybook all the same :(

**Comment by aleksandardjordjeviciohk** [comment]
> > > @aleksandardjordjeviciohk, here is the @alexander-rukin report:
> > > Wrong size:
> > > image (4)
> > > Border issue:
> > > image (3)
> > > Scroll issue:
> > > image (1)
> > 
> > 
> > fixed @alexander-rukin
> 
> what exactly was fixed? in storybook all the same :(

@alexander-rukin pls recheck now, build was broken so changes werent deployed

**Comment by alexander-rukin** [comment]
> @alexander-rukin pls recheck now, build was broken so changes werent deployed

image
search seems to be broken here (+ "X" icon still small)

**Comment by aleksandardjordjeviciohk** [comment]
> > @alexander-rukin pls recheck now, build was broken so changes werent deployed
> 
> image
> 
> search seems to be broken here (+ "X" icon still small)
> UPD: tried local build -> issues still there
> p.s. can you describe issues you are fixing?

Did you tried testing it in the app @alexander-rukin ?

**Comment by aleksandardjordjeviciohk** [comment]
> > @alexander-rukin pls recheck now, build was broken so changes werent deployed
> 
> image
> 
> search seems to be broken here (+ "X" icon still small)
> UPD: tried local build -> issues still there
> p.s. can you describe issues you are fixing?

@alexander-rukin 

Screen Shot 2021-04-14 at 10 06 57 AM

Screen Shot 2021-04-14 at 10 07 04 AM

**Comment by gabriela-ponce** [comment]
@aleksandardjordjeviciohk A few comments about builds 17606:

- This change is implemented in all similar fields, not sure if all these need the search to be enabled, as the list is very short in most cases. Check [screenshot](https://drive.google.com/file/d/1FVfE--CRQag4C2ZK5vGTKy7fc6uWw7nl/view?usp=sharing).
- (Maybe related to the previous item) A blank screen and a console error are displayed when I access "Stake pools". Check [video](https://drive.google.com/file/d/1N2Q-BZsGWoeQdT_0jabocEDfyrbS2Sk5/view?usp=sharing).
- As @alexander-rukin mentioned, the "x" is still small compared to similar buttons on other screens. Check screenshot on "[stake pools](https://drive.google.com/file/d/1vJyenKOBkTs2jOhUAvFXrlUF-bMcJAfl/view?usp=sharing)" and on "[wallets](https://drive.google.com/file/d/1xoMPXnScSfTwYIRI-Aldj-joqEVJoLmW/view?usp=sharing)"

**Comment by aleksandardjordjeviciohk** [comment]
> @aleksandardjordjeviciohk A few comments about builds 17606:
> 
> * This change is implemented in all similar fields, not sure if all these need the search to be enabled, as the list is very short in most cases. Check [screenshot](https://drive.google.com/file/d/1FVfE--CRQag4C2ZK5vGTKy7fc6uWw7nl/view?usp=sharing).
> * (Maybe related to the previous item) A blank screen and a console error are displayed when I access "Stake pools". Check [video](https://drive.google.com/file/d/1N2Q-BZsGWoeQdT_0jabocEDfyrbS2Sk5/view?usp=sharing).
> * As @alexander-rukin mentioned, the "x" is still small compared to similar buttons on other screens. Check screenshot on "[stake pools](https://drive.google.com/file/d/1vJyenKOBkTs2jOhUAvFXrlUF-bMcJAfl/view?usp=sharing)" and on "[wallets](https://drive.google.com/file/d/1xoMPXnScSfTwYIRI-Aldj-joqEVJoLmW/view?usp=sharing)"
> * This is very minor, but a little inconsistent: when searching for special characters, as "." no highlight is displayed. It does happen with other characters, like "-". Check [video](https://drive.google.com/file/d/1P7ihNXCwSpOgmACglHirvopWyDWdCcQ1/view?usp=sharing).

@gabriela-ponce 

Fixed: 
- This change is implemented in all similar fields, not sure if all these need the search to be enabled, as the list is very short in most cases. Check screenshot.

As @alexander-rukin mentioned, the "x" is still small compared to similar buttons on other screens. Check screenshot on "stake pools" and on "wallets"

Fixed in react polymorph (need to wait for the new release in order to test it)
- (Maybe related to the previous item) A blank screen and a console error are displayed when I access "Stake pools". Check video.

Work in progress:
- This is very minor, but a little inconsistent: when searching for special characters, as "." no highlight is displayed. It does happen with other characters, like "-". Check video.

**Comment by aleksandardjordjeviciohk** [comment]
@gabriela-ponce 

Fixed:

- This is very minor, but a little inconsistent: when searching for special characters, as "." no highlight is displayed. It does happen with other characters, like "-". Check video.

This fix is added in react-polymorph PR https://github.com/input-output-hk/react-polymorph/pull/171

**Comment by gabriela-ponce** [comment]
@aleksandardjordjeviciohk the fixes look good on the latest build.
I'll wait for the implementation of these to test on the build:

> - (Maybe related to the previous item) A blank screen and a console error are displayed when I access "Stake pools". Check video.

> - This is very minor, but a little inconsistent: when searching for special characters, as "." no highlight is displayed. It does happen with other characters, like "-".

(The last one is ok in Storybook)

File: package.json
Comment (nikolaglumac) [inline package.json:89]: @aleksandardjordjeviciohk please use fixed version here -> `^1.6.1` should be changed to `1.6.1`

File: source/renderer/app/components/settings/categories/WalletsSettings.scss
Comment (nikolaglumac) [inline source/renderer/app/components/settings/categories/WalletsSettings.scss:90]: @aleksandardjordjeviciohk this will affect ALL em elements within all of the `.SimpleOptions_option` we have in Daedalus. So, either move this declaration elsewhere (e.g. https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/themes/overrides/OptionsOverrides.scss) or make it specific to the context of `.WalletSettings_component`

File: source/renderer/app/themes/overrides/OptionsOverrides.scss
Comment (nikolaglumac) [inline source/renderer/app/themes/overrides/OptionsOverrides.scss:5]: @aleksandardjordjeviciohk I don't think you need to use :global here - hasSearch should be accessible as direct CSS class instead. Please consult with @DominikGuzei

File: storybook/stories/settings/general/General.stories.js
Comment (nikolaglumac) [inline storybook/stories/settings/general/General.stories.js:25]: @aleksandardjordjeviciohk please remove this.

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
@aleksandardjordjeviciohk please implement the changes I have requested 🙏

**Comment by nikolaglumac** [comment]
@aleksandardjordjeviciohk please add a CHANGELOG entry 🙏

**Comment by alexander-rukin** [comment]
This box seems to be shifted to the bottom more (1px) - [screenshot](https://user-images.githubusercontent.com/2539345/114834992-cbf18800-9dd9-11eb-9e11-ae2be9b5ddbd.png)

I'm pretty sure that this whole story with searchable dropdowns was created to be able search [this](https://user-images.githubusercontent.com/2539345/114835812-a3b65900-9dda-11eb-85a4-94f2a4a459b0.png)
Are we going to implement this?

**Comment by nikolaglumac** [comment]
> I'm pretty sure that this whole story with searchable dropdowns was created to be able search [this](https://user-images.githubusercontent.com/2539345/114835812-a3b65900-9dda-11eb-85a4-94f2a4a459b0.png)
> Are we going to implement this?

We will, but in a separate PR.

File: package.json
Comment (aleksandardjordjeviciohk) [reply package.json]: Updated

File: storybook/stories/settings/general/General.stories.js
Comment (aleksandardjordjeviciohk) [reply storybook/stories/settings/general/General.stories.js]: Updated

File: source/renderer/app/components/settings/categories/WalletsSettings.scss
Comment (aleksandardjordjeviciohk) [reply source/renderer/app/components/settings/categories/WalletsSettings.scss]: Makes sense. I have updated it.

File: source/renderer/app/themes/overrides/OptionsOverrides.scss
Comment (aleksandardjordjeviciohk) [reply source/renderer/app/themes/overrides/OptionsOverrides.scss]: Added

**Comment by aleksandardjordjeviciohk** [comment]
> This box seems to be shifted to the bottom more (1px) - [screenshot](https://user-images.githubusercontent.com/2539345/114834992-cbf18800-9dd9-11eb-9e11-ae2be9b5ddbd.png)
> 
> I'm pretty sure that this whole story with searchable dropdowns was created to be able search [this](https://user-images.githubusercontent.com/2539345/114835812-a3b65900-9dda-11eb-85a4-94f2a4a459b0.png)
> Are we going to implement this?

Positioned clear btn @alexander-rukin pls recheck

File: source/renderer/app/components/settings/categories/WalletsSettings.scss
Comment (nikolaglumac) [reply source/renderer/app/components/settings/categories/WalletsSettings.scss]: I don't see the change @aleksandardjordjeviciohk?

Comment (nikolaglumac) [reply source/renderer/app/components/settings/categories/WalletsSettings.scss]: I would rather like to see you move this to the overrides scss...

File: source/renderer/app/themes/overrides/OptionsOverrides.scss
Comment (nikolaglumac) [reply source/renderer/app/themes/overrides/OptionsOverrides.scss]: Can we please do the same for the `.SimpleOptions_search button:before {` 🙏

**Review by alexander-rukin — APPROVED** [review]
Looks ok

File: source/renderer/app/components/settings/categories/WalletsSettings.scss
Comment (DominikGuzei) [inline source/renderer/app/components/settings/categories/WalletsSettings.scss:95]: Please nest the `:global` rule within `.component` selector (then you can remove the hardcoded reference to `.WalletSettings_component` like this:

```scss
.component {
  :global {
      .SimpleOptions_option em {
        background-color: var(--rp-option-search-highlight-background-color) !important;
      }
  }
}
```

same for all other places of course 🙏

**Review by DominikGuzei — CHANGES_REQUESTED** [review]

File: source/renderer/app/components/settings/categories/WalletsSettings.scss
Comment (aleksandardjordjeviciohk) [reply source/renderer/app/components/settings/categories/WalletsSettings.scss]: Fixed

Comment (DominikGuzei) [reply source/renderer/app/components/settings/categories/WalletsSettings.scss:92]: @aleksandardjordjeviciohk at the bottom of this css file is another `:global` that should be refactored the same was 🙏

Comment (aleksandardjordjeviciohk) [reply source/renderer/app/components/settings/categories/WalletsSettings.scss:92]: Fixed

Comment (nikolaglumac) [reply source/renderer/app/components/settings/categories/WalletsSettings.scss:92]: @aleksandardjordjeviciohk this still hasn't been fixed. Please **remove** global declaration from here!

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
@aleksandardjordjeviciohk I have found this error that shows once you enter the "Settings" screen:
Screenshot 2021-04-17 at 19 49 05

...and also I can see that an earlier request has not yet been resolved.

Thanks for fixing those 🙏

File: source/renderer/app/themes/overrides/OptionsOverrides.scss
Comment (aleksandardjordjeviciohk) [reply source/renderer/app/themes/overrides/OptionsOverrides.scss]: Done

File: source/renderer/app/components/settings/categories/WalletsSettings.scss
Comment (nikolaglumac) [reply source/renderer/app/components/settings/categories/WalletsSettings.scss:92]: https://github.com/input-output-hk/daedalus/blob/feature/ddw-645-update-react-polymorph-and-implement-the-search-style-variables/source/renderer/app/components/settings/categories/WalletsSettings.scss#L80-L91

Comment (nikolaglumac) [reply source/renderer/app/components/settings/categories/WalletsSettings.scss:92]: @aleksandardjordjeviciohk please fix these 🙏

Comment (aleksandardjordjeviciohk) [reply source/renderer/app/components/settings/categories/WalletsSettings.scss]: Fixed @nikolaglumac

Comment (daniloprates) [reply source/renderer/app/components/settings/categories/WalletsSettings.scss]: @nikolaglumac @DominikGuzei fixed

**Comment by daniloprates** [comment]
Hey @gabriela-ponce @alexander-rukin @nikolaglumac @DominikGuzei the fixes are done. I've moved 2 of them into React Polymorph's [#175](https://github.com/input-output-hk/react-polymorph/pull/175): [this error](https://user-images.githubusercontent.com/376611/115122135-a4a3e200-9fb6-11eb-93e0-3079960570dc.png) and [this adjustment](https://user-images.githubusercontent.com/1504716/115310248-e0b87d80-a143-11eb-8fca-d0b280e4f479.png).

Temporarily added the PR's branch as dependency version, so it can be tested here. Sasha, Netifly [is here](https://deploy-preview-2512--daedalus-storybook.netlify.app/?path=/story/settings-general--wallets).

**Do not merge it until  [#175](https://github.com/input-output-hk/react-polymorph/pull/175) is merged, [this commit](https://github.com/input-output-hk/daedalus/pull/2512/commits/e3f4f49efb50318586783ae36ac919e721cfba93) is reverted and RP dep. version is updated**.

**Comment by nikolaglumac** [comment]
> Hey @gabriela-ponce @alexander-rukin @nikolaglumac @DominikGuzei the fixes are done. I've moved 2 of them into React Polymorph's [#175](https://github.com/input-output-hk/react-polymorph/pull/175): [this error](https://user-images.githubusercontent.com/376611/115122135-a4a3e200-9fb6-11eb-93e0-3079960570dc.png) and [this adjustment](https://user-images.githubusercontent.com/1504716/115310248-e0b87d80-a143-11eb-8fca-d0b280e4f479.png).
> 
> Temporarily added the PR's branch as dependency version, so it can be tested here. Sasha, Netifly [is here](https://deploy-preview-2512--daedalus-storybook.netlify.app/?path=/story/settings-general--wallets).
> 
> **Do not merge it until [#175](https://github.com/input-output-hk/react-polymorph/pull/175) is merged, [this commit](https://github.com/input-output-hk/daedalus/pull/2512/commits/e3f4f49efb50318586783ae36ac919e721cfba93) is reverted and RP dep. version is updated**.

@daniloprates please let me know once this is ready to be merged.
FYI. You won't be getting Linux or Windows builds as long as you are setting the git commit as a version for RP (I know this is temporary - just wanted to point out we have only macOS builds at the moment...)

File: package.json
Comment (DominikGuzei) [inline package.json:237]: Does it work to reference commits with Daedalus builds nowadays?

Comment (nikolaglumac) [reply package.json:237]: No. This is only temporary. https://github.com/input-output-hk/daedalus/pull/2512#issuecomment-823031379

Comment (DominikGuzei) [reply package.json:237]: I will make a new RP release for this PR

**Comment by DominikGuzei** [comment]
@daniloprates i merged https://github.com/input-output-hk/react-polymorph/pull/175 and released it as `react-polymorph@0.9.8-rc.20`

**Comment by daniloprates** [comment]
> @daniloprates i merged input-output-hk/react-polymorph#175 and released it as react-polymorph@0.9.8-rc.20

@DominikGuzei great, thanks 🚀 

> FYI. You won't be getting Linux or Windows builds as long as you are setting the git commit as a version for RP

@nikolaglumac thanks for the heads up

@gabriela-ponce @alexander-rukin could you give a last run on this? 🙏 

([Netlify](https://deploy-preview-2512--daedalus-storybook.netlify.app/?path=/story/settings-general--wallets))

**Comment by gabriela-ponce** [comment]
@daniloprates Can we only test this on Netlify? I downloaded the latest build on this PR and [some inputs were missing on Settings](https://drive.google.com/file/d/1XcbsJc4OhXxpZuEaDnTW4yPCRJ3z1sGd/view?usp=sharing).

**Comment by daniloprates** [comment]
@gabriela-ponce found it's not working, so @alexander-rukin don't waste time, right now.

@DominikGuzei it seems `0.9.8-rc.20` is not working. If I check with `develop` it does, though. Maybe it's missing building or something like this?

**Review by alexander-rukin — CHANGES_REQUESTED** [review]
Now cross again small 
image

**Review by nikolaglumac — APPROVED** [review]

**Review by DominikGuzei — APPROVED** [review]
Looks good 🎉

**Comment by alexander-rukin** [comment]
image

Two things I see: 
- search is not cleared after selection 
- there is no "selection checkmark" here

**Comment by daniloprates** [comment]
@alexander-rukin this is because of Storybook, my changes don't touch these parts

**Review by gabriela-ponce — APPROVED** [review]

**Comment by daniloprates** [comment]
> * search is not cleared after selection
> * there is no "selection checkmark" here

@alexander-rukin these 2 things need to be done on React-Polymorph. Please, approve this PR, as it's only for the styling. I'll create a RP card for these ☝️

**Review by alexander-rukin — APPROVED** [review]

**Pull request merged**
